beacon_node/network/src/sync/block_lookups/mod.rs

```diff
@@ -759,90 +435,92 @@ impl<T: BeaconChainTypes> BlockLookups<T> {
-                        let slot = block.slot();
-                        let parent_root = block.parent_root();
-                        lookup.add_child_components(block.into());
-                        Action::ParentUnknown { parent_root, slot }
+                        // Reverts the status of this request to `AwaitingProcessing` holding the
+                        // downloaded data. A future call to `continue_requests` will re-submit it
+                        // once there are no pending parent requests.
+                        // Note: `BlockError::ParentUnknown` is only returned when processing
+                        // blocks, not blobs.
+                        request_state.revert_to_awaiting_processing()?;
+                        Action::ParentUnknown {
+                            parent_root: block.parent_root(),
+                        }
                     }
                     ref e @ BlockError::ExecutionPayloadError(ref epe) if !epe.penalize_peer() => {
                         // These errors indicate that the execution layer is offline
                         // and failed to validate the execution payload. Do not downscore peer.
                         debug!(
                             self.log,
                             "Single block lookup failed. Execution layer is offline / unsynced / misconfigured";
-                            "root" => %root,
+                            "block_root" => %block_root,
                             "error" => ?e
                         );
                         Action::Drop
                     }
-                    BlockError::AvailabilityCheck(e) => match e.category() {
-                        AvailabilityCheckErrorCategory::Internal => {
-                            warn!(self.log, "Internal availability check failure"; "root" => %root, "peer_id" => %peer_id, "error" => ?e);
-                            lookup.block_request_state.state.on_download_failure();
-                            lookup.blob_request_state.state.on_download_failure();
-                            Action::Retry
-                        }
-                        AvailabilityCheckErrorCategory::Malicious => {
-                            warn!(self.log, "Availability check failure"; "root" => %root, "peer_id" => %peer_id, "error" => ?e);
-                            lookup.handle_availability_check_failure(cx);
-                            Action::Retry
-                        }
-                    },
+                    BlockError::AvailabilityCheck(e)
+                        if e.category() == AvailabilityCheckErrorCategory::Internal =>
+                    {
+                        // There errors indicate internal problems and should not downscore the  peer
+                        warn!(self.log, "Internal availability check failure"; "block_root" => %block_root, "error" => ?e);
+                        // TODO: This lines represent an improper transition of download states,
```

Comment:
Ok will try to illustrate with an example, see you can help me get a better comment:

- Lookup created
- Block request sent
- Blob request sent
- Blob request completed, and sent for processing (blob request state = Processing)
- Block request completed, sent for processing
- Block processing returns `BlockError::AvailabilityCheck(AvailabilityCheckErrorCategory::Internal)`
- Call `blob.on_download_failure()` results in `bad state: expected Downloading got Processing`

This error can happen with different variations of this scenario as you don't know the status of the blob request which is happening in parallel. You can error the block request because you know it is in `Processing` state, but not the blob request.

If we are currently forcing the blob request to transition from `Processed` into `AwaitingDownload` it means that a `AvailabilityCheckErrorCategory::Internal` can invalidate blobs that were previously considered valid, which is very weird. I don't understand under what circumstances that would happen. If there are concerns that an `AvailabilityCheckErrorCategory::Internal` break consistency or it is not recoverable we should just drop the request.

Reply:
Thanks for this description, i had my hand at writing a comment. does this seem accurate to you? 3544fbde94bba56a8560b5223a027803383589b2